solutions/5.2.piece_of_cake.py

```diff
@@ -0,0 +1,27 @@
+def get_recipe_price(prices, optionals=None, **kwargs):
+    """
+    Calculate the total cost of a recipe based on the ingredient prices and quantities specified.
+
+    Args:
+        prices (dict): A dictionary of ingredient prices (per 100 grams).
+        optionals (list, optional): A list of optional ingredients to be excluded from the cost calculation.
+                                    Defaults to an empty list.
+        **kwargs (float): The quantities of each ingredient used in the recipe, specified as keyword arguments,
+                           where the key is the ingredient name (str) and the value is the quantity (float).
+
+    Returns:
+        The total cost of the recipe, based on the ingredient prices and quantities specified.
+    """
+    if prices == {}:
+        return 0
+
+    if optionals is None:
+        optionals = []
+
+    total_sum = 0
+
+    for ingredient, quantity in kwargs.items():
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.

Reply:
done